http2/server.go

```diff
@@ -1908,7 +1908,7 @@ func (sc *serverConn) processHeaders(f *MetaHeadersFrame) error {
 	// technically more like the http1 Server's ReadHeaderTimeout
 	// (in Go 1.8), though. That's a more sane option anyway.
 	if sc.hs.ReadTimeout != 0 {
-		sc.conn.SetReadDeadline(time.Time{})
+		sc.conn.SetReadDeadline(time.Now().Add(sc.hs.ReadTimeout))
```

Comment:
Per the comment above it seems that disarming the timeout here is on purpose,

Reply:
my misunderstanding for ReadTimeout
btw, here's need to err = sc.conn.SetReadDeadline() ....